Release Notes for v1.0.0

trbehera opened the pull request:
Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera committed:
Release Notes for v1.0.0

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera committed:
Added supported crypto, known issues

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

prateekcr commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+
+* For COSE signatures, ES512 signature type is not supported. (section 3.3.5)
+
+**client-sdk-fidoiot**: It supports a only a subset of the cryptographic modes.
```
Typo "a only" -> "only"

trbehera replied:
Fixed

prateekcr commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+**pri-fidoiot**: RVDelaySec is not honored for TO0 retries with RendezvousInfo, due to possible
+conflicts with the TO0Scheduler.
+
+**client-sdk**: IPAddress.ip6 is not supported. (section 3.3.9)
```
client-sdk-fidoiot

trbehera replied:
Fixed

trbehera commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+  * EPID Verification Service: [epid-verification-service](https://github.com/secure-device-onboard/epid-verification-service) is a wrapper service written on top of EPID SDK to assist FDO Rendezvous service and FDO Owner service to perform device signature verification for EPID based devices.
+  * Test: [test-fidoiot](https://github.com/secure-device-onboard/test-fidoiot) implements a test-suite that gets executed as part of continuous integration pipeline.
+
+### New Features
```
Change the order

==> Remove heading "Component"

New Features
Changes to existing features

Supported Crypto Modes
Known Issues
==> Move implementation notes into Known Issues

DukeDavis12 commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+
+### Supported Cryptographic Modes
+
+**pri-fidoiot** supports following cryptographic modes.
```
supports following ==> supports the following

DukeDavis12 commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+
+* For COSE signatures, ES512 signature type is not supported. (section 3.3.5)
+
+**client-sdk-fidoiot**: It supports a only a subset of the cryptographic modes.
```
It supports a only a ==> It supports only a

trbehera replied:
Fixed

DukeDavis12 commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+* For key exchange, it supports DHKEXid14, DHKEXid15, ASYMKEX2048, AESYMKEX3072, ECDH256 and ECDH384
+  modes.
+
+* For session encryption, it supports following AES modes - AES-CTR-128, AES-CTR-256, AES-CBC-128,
```
supports following ==> supports the following

trbehera replied:
Fixed

DukeDavis12 commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+
+* For Key Exchange, ECDH256 and ECDH384 modes are supported.
+
+* For session encryption, it supports following AES modes - A128GCM, A256GCM, AES-CCM-64-128-128 and
```
supports following ==> supports the following

trbehera replied:
Fixed

DukeDavis12 commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+provided, a delay of 3 seconds is used while trying successive RendezvousDirectives and a delay of
+120 seconds is used while retrying the RendezvousInfo. The delay is not randomized by +/-30s.
+
+**pri-fidoiot**: RVDelaySec is not honored for TO0 retries with RendezvousInfo, due to possible
```
honored ==> honoured

trbehera replied:
Fixed

johneasterday commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+
+It includes 4 components:
+  * Protocol Reference Implementation (PRI): [pri-fidoiot](https://github.com/secure-device-onboard/pri-fidoiot) is a JAVA based implementation of all the components specified in the FDO Specification.
+  * Client SDK: [client-sdk-fidoiot](https://github.com/secure-device-onboard/client-sdk-fidoiot) is a C based implementation for the device component specified in the FDO Specification. Additionally, it supports an implementation of the device that uses TPM infrastructure.
```
"uses TPM" -> "uses the TPM"

trbehera replied:
Fixed

johneasterday commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+It includes 4 components:
+  * Protocol Reference Implementation (PRI): [pri-fidoiot](https://github.com/secure-device-onboard/pri-fidoiot) is a JAVA based implementation of all the components specified in the FDO Specification.
+  * Client SDK: [client-sdk-fidoiot](https://github.com/secure-device-onboard/client-sdk-fidoiot) is a C based implementation for the device component specified in the FDO Specification. Additionally, it supports an implementation of the device that uses TPM infrastructure.
+  * EPID Verification Service: [epid-verification-service](https://github.com/secure-device-onboard/epid-verification-service) is a wrapper service written on top of EPID SDK to assist FDO Rendezvous service and FDO Owner service to perform device signature verification for EPID based devices.
```
"top of EPID" -> "top of the EPID"
"assist FDO Rendezvous" -> "assist the FDO Rendezvous"

trbehera replied:
Fixed

johneasterday commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+
+**pri-fidoiot** supports following cryptographic modes.
+
+* For device attestation, it supports Intel EPID keys based on StEPID10 and StEPID11 and ECC keys
```
What does the St mean in "StEPID10" and "StEPID11" ?

trbehera replied:
EPID 1.0 and EPID 1.1

trbehera replied:
Fixed

johneasterday commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+* For key exchange, it supports DHKEXid14, DHKEXid15, ASYMKEX2048, AESYMKEX3072, ECDH256 and ECDH384
+  modes.
+
+* For session encryption, it supports following AES modes - AES-CTR-128, AES-CTR-256, AES-CBC-128,
```
"support following" -> "supports the following"

trbehera replied:
Fixed

johneasterday commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,110 @@
+**pri-fidoiot**, **cient-sdk-fidoiot**: The following RVProtocolValue entries are ignored -
+RVProtRest, RVProtTcp, RVProtCoapTcp and RVProtCoapUdp.
+
+**client-sdk-fidoiot**: Maximum supported values for RVDelaySec is 3600s. If no RVDelaySec is
```
"support values" -> "supported value"

trbehera replied:
Fixed

trbehera committed:
Additional changes

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,141 @@
+**pri-fidoiot**: A new component 'aio' (All-in-one Demo) is included. It implements all the FDO
+services witin a single container. REST APIs are provided to configure the component.
+
+**pri-fidoiot**: Docker based build scripts are provided for building the source code.
```
include epid for build infra update

trbehera replied:
Fixed

trbehera commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,141 @@
+
+### Changes to existing features
+
+**pri-fidoiot**: Diffie-Hellman key exchage support is added.
```
exchange

trbehera replied:
Fixed

trbehera commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,141 @@
+
+* For public key encoding, it supports Crypto, X509 as well as COSEX509 encoding. (section 3.3.4)
+
+* For COSE signatures, ES512 signature type is not supported. (section 3.3.5)
```
Add specifically what is supported.

trbehera replied:
Fixed

trbehera commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,141 @@
+
+* For public key encoding, COSEX509 encoding is used. (section 3.3.4).
+
+* For COSE signatures, ES512, RS256 and RS384 signature types are not supported. (section 3.3.5)
```
Add explicitly what is supported.

trbehera replied:
Fixed

prateekcr commented on release-notes-v1.0.0.md:
```diff
@@ -0,0 +1,141 @@
+
+### Known Issues
+
+**client-sdk-fidoiot**: The device ServiceInfo message is not broken down into multiple messages
```
This is being done now except for one key. Instead of this, please add the following info:
1. The key 'devmod:modules' and its corresponding value cannot be broken into multiple message based on the maximum MTU size indicated by the Owner, and is sent in a single message. Addition of ServiceInfo modules to this key requires manual change in the source.
Also,
2. Only the 'devmod' module is supported as a Device ServiceInfo module. Support for adding additional Device ServiceInfo modules and subsequent response framework to the Owner, has not yet been implemented,

trbehera replied:
Fixed

trbehera committed:
Update as per review comments

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera committed:
Update known issues links

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera committed:
Fold in crypto support to features

Rolled in the section for "Supported Cryptographic Features" into "New
Features".

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera committed:
Update README file

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

trbehera committed:
Reorder the entries for different components

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>
